client/dashboard/sites/deployments/index.tsx

```diff
@@ -1,16 +1,72 @@
+import { useQuery } from '@tanstack/react-query';
+import { __experimentalText as Text } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
+import { siteQuery } from '../../app/queries';
+import { siteRoute } from '../../app/router';
+import { Callout } from '../../components/callout';
+import { CalloutOverlay } from '../../components/callout-overlay';
 import DataViewsCard from '../../components/dataviews-card';
 import { PageHeader } from '../../components/page-header';
 import PageLayout from '../../components/page-layout';
+import RouterLinkButton from '../../components/router-link-button';
+import { SitePlan } from '../../data/types';
+import illustrationUrl from './deployments-callout-illustration.svg';
+import ghIconUrl from './gh-icon.svg';
 
 function SiteDeployments() {
+	const { siteSlug } = siteRoute.useParams();
+	const { data: site } = useQuery( siteQuery( siteSlug ) );
+
+	if ( ! site ) {
+		return;
+	}
+
+	const showCallout =
+		( ! site.plan || ! hasDeploymentsFeature( site.plan ) ) &&
+		// Show to all atomic sites regardless of plan status e.g. an atomic site in the revert queue
+		// after a plan downgrade will continue to see Deployments until the revert is complete
+		! site.is_wpcom_atomic;
+
 	return (
 		<PageLayout header={ <PageHeader title={ __( 'Deployments' ) } /> }>
-			<DataViewsCard>
-				<></>
-			</DataViewsCard>
+			<CalloutOverlay
+				showCallout={ showCallout }
+				callout={
+					<Callout
+						icon={ <img src={ ghIconUrl } alt={ __( 'GitHub logo' ) } /> }
+						title={ __( 'Deploy from GitHub' ) }
+						imageSrc={ illustrationUrl }
+						description={
+							<>
+								<Text as="p" variant="muted">
+									{ __(
+										'Connect your GitHub repo directly to your WordPress.com site—with seamless integration, straightforward version control, and automated workflows.'
+									) }
+								</Text>
+								<Text as="p" variant="muted">
+									{ __( 'Available on the WordPress.com Business and Commerce plans.' ) }
+								</Text>
+							</>
+						}
+						actions={
+							<RouterLinkButton __next40pxDefaultSize variant="primary" to="#">
+								{ __( 'Upgrade plan' ) }
+							</RouterLinkButton>
+						}
+					/>
+				}
+				main={
+					<DataViewsCard>
+						<></>
+					</DataViewsCard>
```

Comment:
I'm not sure how this works. Shouldn't the content of `main` depends on `showCallout`? As in, if `showCallout` is true, we should render a blurred placeholder content here; otherwise, we render the real component (the real Deployments page). How would we handle that?

Reply:
Oooh that's an interesting point. I was imagining the placeholder content would use the same elements as the real page, and it would be the data that was swapped out for dummy data. But I see now that the placeholder content could be totally different elements.

I like the way that in this situation the placeholder is using the same elements as the live data. But shall we wait until we have some placeholder components before we settle on what the right abstraction is? The main thing about this PR is that the `CalloutOverlay` needs to own the placeholder elements so that it can correctly blur them.

Reply:
> I see now that the placeholder content could be totally different elements.

Yeah, this is what I imagine, too:

- Deployments renders deployment list with many rows of fake data
- Monitoring renders fake graphs

> placeholder content would use the same elements as the real page, and it would be the data that was swapped out for dummy data.

Not sure but I *think* it's harder to pass around the `showCallout` state to various downstream components, instead of just coming up with a fake static content in the first place.

---

Not a blocker though, we can change the logic later, just wanted to call out (pun slightly intended)!